app/models/ems_event.rb

```diff
@@ -146,6 +146,34 @@ def self.process_vm_in_event!(event, options = {})
         event[:vm_name] ||= vm.name
       end
     end
+
+    event_type = event[:event_type]
+    # both event types are Rhev specific
+    if (event_type == 'USER_ADD_VM' || event_type == 'USER_ADD_VM_FINISHED_SUCCESS') && event[:vm_or_template_id].nil?
```

Comment:
So for RHEV a VM won't have a host assigned yet so we'd have to move up to the cluster